### ISSUE TITLE
Remove non exhaustive list binding from docs

### ISSUE THREAD

**Issue by amiralies** [reason-association/rescript-lang.org#198]
From docs:
https://rescript-lang.org/docs/manual/latest/pattern-matching-destructuring

```rescript
// List
let myList = list{1, 2, 3}
let list{head, ...tail} = myList // 1 assigned to `head`, `list{2, 3}` assigned to tail
```
Compiler currently crashes in such case.
but aside that, since the the pattern fails for an empty list  I think we should remove this or warn the user about that.

**Comment by kevanstannard** [comment]
Since the example is documenting pattern matching destruction for lists (and arrays above it) what do you think about updating the example rather than removing it?

For example:

```res
let myList = list{1, 2, 3}

let (head, tail) = switch myList {
| list{head, ...tail} => (Some(head), tail)
| list{} => (None, list{})
}
```

**Comment by amiralies** [comment]
This is too complex in the context of destructuring IMO. I prefer adding a note explaining it can be fail and the pattern is not exhaustive.

**Comment by kevanstannard** [comment]
Understood, I see your concern.

What about moving that content to a separate section "Destructuring Arrays and Lists" which includes examples like above.

Including a note explaining that list/array structuring can fail and that the pattern not being exhaustive is good information, but it doesn't help people new to ReScript deal with that problem. And I imagine those people are a primary target audience for the docs.

In other words, if we describe a problem then it would be good to also provide a guide to resolve it (either a link to another page, or describe the solution in the same place).

Also, maybe there is a simpler example we could use?

What do you think?

**Comment by ryyppy** [comment]
This example should now be documented properly in master:

![image](https://user-images.githubusercontent.com/1121889/106566967-58a5dd00-6531-11eb-824d-8059b1f2a1e0.png)


I think we can close this issue?

**Comment by amiralies** [comment]
Yup, Thank you.
